components/browser/storage-sync/src/main/java/mozilla/components/browser/storage/sync/RemoteTabsStorage.kt

```diff
@@ -0,0 +1,145 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.storage.sync
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.cancelChildren
+import kotlinx.coroutines.withContext
+import mozilla.appservices.remotetabs.RemoteTab
+import mozilla.appservices.remotetabs.RemoteTabProviderException
+import mozilla.components.concept.storage.Storage
+import mozilla.components.concept.sync.SyncableStore
+import mozilla.components.support.base.log.logger.Logger
+import mozilla.appservices.remotetabs.RemoteTabsProvider
+import mozilla.appservices.remotetabs.SyncAuthInfo as RustSyncAuthInfo
+import mozilla.components.concept.sync.SyncAuthInfo
+import mozilla.components.concept.sync.SyncStatus
+
+/**
+ * An interface which defines read/write methods for remote tabs data.
+ */
+open class RemoteTabsStorage : Storage, SyncableStore {
+    internal val api by lazy { RemoteTabsProvider() }
+    private val scope by lazy { CoroutineScope(Dispatchers.IO) }
+    internal val logger = Logger("PlacesHistoryStorage")
+
+    /**
+     * Store the locally opened tabs.
+     * @param tabs The list of opened tabs, for all opened non-private windows, on this device.
+     */
+    fun store(tabs: List<Tab>) {
```

Comment:
Needs to be a suspend function.

Reply:
Android Studio tells me it's redundant because nothing `suspend` is being called in there (the rust calls are synchronous as there's no network requests/io being made).

Reply:
The `api` calls need to be wrapped in a coroutine so that they aren't done on the UI thread. If we execute this as is, it would be calling the native stack with reading to disk, making networks calls, and this would block the UI. Have a look at the PlacesBookmarkStorage for examples on how to make this async.